tests/test_document.py

```diff
@@ -58,6 +59,19 @@
         await doc2.save()
 
 
+async def test_conflict_without_rev(filled_database):
+    doc1 = Document(filled_database, "fou")
+    doc2 = Document(filled_database, "fou")
```

Comment:
```suggestion
    doc1 = await database.create("fou")
    doc2 = await database.create("fou")
```
I'd prefer that the tests use the public API if possible.

Reply:
The point of this test is to wrap `HTTP 409` on a doc that was never accessed from the network (no `create()`, no `fetch()`, no `_exists()`. That's why I'd prefer keeping it like this. What do you think?

By the way, if calling `Document()` isn't the "public API", is there a way to create such an object, without issuing any call to CouchDB?

Reply:
> The point of this test is to wrap `HTTP 409` on a doc that was never accessed from the network (no `create()`, no `fetch()`, no `_exists()`, no `_rev`). That's why I'd prefer keeping it like this. What do you think?

Sorry, maybe my wording was a bit off here. For the time being, I have to acknowledge that there is a significant lack of documentation for `aiocouch`. Therefore, the next best thing to learn the usage of it is to look at the examples and tests. Unfortunately, there aren't many examples, either. Thus, I'd like at least the test to be written in the recommended way. The recommended way to getting an instance of the Document class is to call `Database.create()`, if the document (probably) doesn't exist yet. Otherwise, that is `Database.get()`. I'm aware that those functions trigger a maybe unnecessary HEAD or GET request, but they will catch unexpected situations. The side-effect of using those methods is that it is unlikely to break the behavior of them in the future. Side note, `create` doesn't really create a new document on the server, only the `save()` on the returned object will.

> By the way, if calling `Document()` isn't the "public API", is there a way to create such an object, without issuing any call to CouchDB?

It's poorly worded, read it as "recommended API." I still recommend using `create` or `get`, respectively. But if you are concerned about the additional request and you happen to know that the document you want to create cannot exist yet then sure, go for it.

Reply:
> I'm aware that those functions trigger a maybe unnecessary HEAD or GET request, but they will catch unexpected situations. The side-effect of using those methods is that it is unlikely to break the behavior of them in the future.

OK, I understand! I changed to PR to use your suggestion.

> But if you are concerned about the additional request and you happen to know that the document you want to create cannot exist yet then sure, go for it.

Yes, in our code we have to use a lot of these:
```python
# This document will be immutable, we just want to create it if not existing
# yet; using only one HTTP call for performance reasons.
doc = Document(Database(couch, 'db'), 'id')
doc.update({'key': 'val'})
try:
    await doc.save()
except aiocouch.exception.ConflictError:
    pass
```
... it would be handy to be able to replace it with:
```python
doc = Database(couch, 'db').create('id', do_a_http_head_or_get=False, default={'key': 'val'})
await doc.save(ignore_conflict=True)
```

Reply:
Thanks. I'll merge it.

We should discuss the handling of ConflictErrors in a different issue. Can you open a new issue and provide common cases you have identified in your code?

Reply:
Sure, I'll do it.

In the meantime, is it possible to get a 1.0.1? Can I help in any way?

Reply:
Thank you @bmario!